pkg/provider/aws/plugin.go

```diff
@@ -96,7 +125,9 @@ func (a *AWS) CopyConstructEdgesToDag(result *core.ConstructGraph, dag *core.Res
 				data.Destination = targetTG
 			}
 		}
-		dag.AddDependencyWithData(sourceResource, targetResource, data)
+		if sourceResource != targetResource {
+			dag.AddDependencyWithData(sourceResource, targetResource, data)
+		}
```

Comment:
This check (source != target) should be checked in the DAG code probably.

Reply:
makes sense, ill move it over